### PR TITLE
Agglomerative Clustering 

### PR DESCRIPTION
**Pull request by hcho1111**
- Adding agglomerative clustering plot and heuristics
- Added relevant test modeling idealized outcome, reference `baseline_images` directory

### CODE REVIEW

File: src/sklearn_evaluation/plot/agglomerative.py
Comment (edublancas) [inline src/sklearn_evaluation/plot/agglomerative.py:10]: please change this to an absolute import `from sklearn.telemetry import SKLearnEvaluationLogger`

Comment (edublancas) [reply src/sklearn_evaluation/plot/agglomerative.py:10]: you'll see some relative imports in the codebase but we're moving away from them

File: src/sklearn_evaluation/plot/agglomerative.py
Comment (edublancas) [inline src/sklearn_evaluation/plot/agglomerative.py:109]: this looks similar to the elbow curve (https://sklearn-evaluation.readthedocs.io/en/latest/api/plot.html#sklearn_evaluation.plot.elbow_curve), is the only difference that this is in the PCA space? Let's ensure we document how is this different from the elbow curve so users know when to use which one

**Review by edublancas — CHANGES_REQUESTED** [review]
Overall it looks great! If you want another review, just let me know. Once the final version is in, ping me

File: src/sklearn_evaluation/plot/agglomerative.py
Comment (edublancas) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: also, check the CI, one test is failing

**Comment by idomic** [comment]
@hcho1111 please fix issues, make sure the CI is green and tag the team for review.

File: src/sklearn_evaluation/plot/agglomerative.py
Comment (WSShawn) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: Hi! I'm trying to fix this issue right now but I'm not sure where exactly in the document should I separate this from the elbow curve. Does it mean that we want a separate section on the page that is different from the elbow_curve section?

Comment (WSShawn) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: Working on separating the plot from elbow_curve on the document page. But I'm not sure which file I should edit to make changes on the document page. Please let me know.

Comment (edublancas) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: I think we can just describe the difference with respect to the elbow curve in the docstring. but I think more  importantly, we should know if this is *the same* as the elbow curve, just with a different metric. in such case, we'd want to delete this one so we don't have duplicates.

any hints @hcho1111 ?

Comment (WSShawn) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: So for now, I guess I can start by adding the commits with fixes and open a PR. For the docstring, I'm still slightly confused in terms of judging how different this is from the elbow_curve.

Comment (idomic) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: @WSShawn let's wait for harisson's response since it will affect implementation.
Focus on other issues for now.

Comment (WSShawn) [reply src/sklearn_evaluation/plot/agglomerative.py:109]: ok！

**Pull request closed without merging**
